KSystemInformer/protection.c

```diff
@@ -710,26 +712,29 @@ VOID KphApplyObProtections(
 
         allowedAccessMask = process->ThreadAllowedMask;
 
-        if (KphpShouldPermitCreatorProcess(Info, actor, process))
-        {
-            allowedAccessMask |= (KPH_THREAD_READ_ACCESS |
-                                  THREAD_TERMINATE |
-                                  THREAD_RESUME);
-
-            if ((allowedAccessMask & process->ThreadAllowedMask)
-                != allowedAccessMask)
-            {
-                KphTracePrint(TRACE_LEVEL_VERBOSE,
-                              PROTECTION,
-                              "Permitting extra thread handle access "
-                              "(0x%08x -> 0x%08x) in creator process %lu for "
-                              "process %lu",
-                              process->ThreadAllowedMask,
-                              allowedAccessMask,
-                              HandleToULong(actor->ProcessContext->ProcessId),
-                              HandleToULong(process->ProcessId));
-            }
-        }
+        //if (KphpShouldPermitCreatorProcess(Info, actor, process))
```

Comment:
This code's existence is intentional for the same reasons mentioned in the other comment.

Given `KPH_PROTECTED_THREAD_MASK` contains all the bits in that would be applied for the creator process, it's technically fine to remove this block - however, I stand firm on my principals for design here (as explained in the other comment). This would in fact save a call and some instructions. However, I firmly believe the design reasoning of the implementation outweighs the minuscule cost savings.

Reply:
Also noteworthy, with this change, we lost some trace information - I think it's worth keeping the code the way it was for that reason too.